tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedEventMatcher.cs

```diff
@@ -0,0 +1,167 @@
+{
+    public class UnifiedEventMatcher
+    {
+        private UnifiedValueMatcher _valueMatcher;
+
+        public UnifiedEventMatcher(UnifiedValueMatcher valueMatcher)
+        {
+            _valueMatcher = valueMatcher;
+        }
+
+        public void AssertEventsMatch(List<object> actualEvents, BsonArray expectedEventsDocuments)
+        {
+            try
+            {
+                AssertEvents(actualEvents, expectedEventsDocuments);
+            }
+            catch (XunitException exception)
+            {
+                throw new AssertionException(
+                    userMessage: GetAssertionErrorMessage(actualEvents, expectedEventsDocuments),
+                    innerException: exception);
+            }
+        }
+
+        // private methods
+        private void AssertEvents(List<object> actualEvents, BsonArray expectedEventsDocuments)
+        {
+            actualEvents.Should().HaveSameCount(expectedEventsDocuments);
+
+            for (int i = 0; i < actualEvents.Count; i++)
+            {
+                var actualEvent = actualEvents[i];
+                var expectedEventDocument = expectedEventsDocuments[i].AsBsonDocument;
+                if (expectedEventDocument.ElementCount != 1)
+                {
+                    throw new FormatException("Expected event document model must contain a single element");
+                }
+                var expectedEventType = expectedEventDocument.GetElement(0).Name;
+                var expectedEventValue = expectedEventDocument[0].AsBsonDocument;
+
+                switch (actualEvent)
+                {
+                    case CommandStartedEvent commandStartedEvent:
+                        expectedEventType.Should().Be("commandStartedEvent");
+                        foreach (var element in expectedEventValue)
+                        {
+                            switch (element.Name)
+                            {
+                                case "command":
+                                    _valueMatcher.AssertValuesMatch(commandStartedEvent.Command, element.Value);
+                                    break;
+                                case "commandName":
+                                    commandStartedEvent.CommandName.Should().Be(element.Value.AsString);
+                                    break;
+                                case "databaseName":
+                                    commandStartedEvent.DatabaseNamespace.DatabaseName.Should().Be(element.Value.AsString);
+                                    break;
+                                default:
+                                    throw new FormatException($"Unexpected commandStartedEvent field: '{element.Name}'");
+                            }
+                        }
+                        break;
+                    case CommandSucceededEvent commandSucceededEvent:
+                        expectedEventType.Should().Be("commandSucceededEvent");
```

Comment:
Replace 88-89 with:

    case "commandSucceededEvent":
        var commandSucceededEvent = actualEvent.Should().BeOfType<CommandSucceededEvent>().Subject;